src/main/notebooks/TPC-multi_datagen.scala

```diff
@@ -0,0 +1,277 @@
+    s"$baseLocation/tpcds/sf$scaleFactor-$fileFormat/useDecimal=false,useDate=false,filterNull=false")
+}
+
+// COMMAND ----------
+
+// Data generation
+def isPartitioned (tables: Tables, tableName: String) : Boolean = 
+  util.Try(tables.tables.find(_.name == tableName).get.partitionColumns.nonEmpty).getOrElse(false)
+
+def loadData(tables: Tables, location: String, scaleFactor: String) = {
+  val tableNames = tables.tables.map(_.name)
+  tableNames.foreach { tableName =>
+  // generate data
+    time {
+      tables.genData(
+        location = location, 
+        format = fileFormat, 
+        overwrite = overwrite, 
+        partitionTables = true, 
+        // if to coallesce into a single file (only one writter for non partitioned tables = slow) 
+        clusterByPartitionColumns = shuffle, //if (isPartitioned(tables, tableName)) false else true, 
+        filterOutNullPartitionValues = false, 
+        tableFilter = tableName,
+        // this controlls parallelism on datagen and number of writers (# of files for non-partitioned)
+        // in general we want many writers to S3, and smaller tasks for large scale factors to avoid OOM and shuffle errors    
+        numPartitions = if (scaleFactor.toInt <= 100 || !isPartitioned(tables, tableName)) (workers * cores) else (workers * cores * 4))
+    }
+  }
+}
+
+// COMMAND ----------
+
+// Create the DB, import data, create
+def createExternal(location: String, dbname: String, tables: Tables) = {
+  tables.createExternalTables(location, fileFormat, dbname, overwrite = overwrite, discoverPartitions = true)
+}
+
+def loadDB(dbname: String, tables: Tables, location: String) = {
+  val tableNames = tables.tables.map(_.name)
+  time { 
+    println(s"Creating external tables at $location")
+    createExternal(location, dbname, tables) 
+  }
+  // Show table information and attempt to vacuum
+  tableNames.foreach { tableName =>
+    println(s"Table $tableName has " + util.Try(sql(s"SHOW PARTITIONS $tableName").count() + " partitions").getOrElse(s"no partitions"))
+    util.Try(sql(s"VACUUM $tableName RETAIN 0.0. HOURS"))getOrElse(println(s"Cannot VACUUM $tableName"))
+    sql(s"DESCRIBE EXTENDED $tableName").show(999, false)
+    println
+  }
+}
+
+// COMMAND ----------
+
+def setScaleConfig(scaleFactor: String): Unit = {
+  // Avoid OOM when shuffling large scale fators
+  // and errors like 2GB shuffle limit at 10TB like: Most recent failure reason: org.apache.spark.shuffle.FetchFailedException: Too large frame: 9640891355
+  // For 10TB 16x4core nodes were needed with the config below, 8x for 1TB and below. 
+  // About 24hrs. for SF 1 to 10,000.
+  if (scaleFactor.toInt >= 10000) {    
+    spark.conf.set("spark.sql.shuffle.partitions", "20000")
+    SparkHadoopUtil.get.conf.set("parquet.memory.pool.ratio", "0.1")
+  } 
+  else if (scaleFactor.toInt >= 1000) {
+    spark.conf.set("spark.sql.shuffle.partitions", "2001") //one above 2000
```

Comment:
why? :)

Reply:
To use `HighlyCompressedMapStatus` https://github.com/apache/spark/blob/4be33758354e1f95fd1d82a5482f3f00218e8c91/core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala#L52-L58